instrumentation/active_support/lib/opentelemetry/instrumentation/active_support/span_subscriber.rb

```diff
@@ -29,18 +32,36 @@ def self.subscribe(
         )
 
         subscriber_object = ::ActiveSupport::Notifications.subscribe(pattern, subscriber)
+
         ::ActiveSupport::Notifications.notifier.synchronize do
-          if ::Rails::VERSION::MAJOR >= 6
-            s = ::ActiveSupport::Notifications.notifier.instance_variable_get(:@string_subscribers)[pattern].pop
-            ::ActiveSupport::Notifications.notifier.instance_variable_get(:@string_subscribers)[pattern].unshift(s)
+          subscribers = find_subscribers(pattern)
+
+          if subscribers.nil?
+            OpenTelemetry.handle_error(
+              message: 'Unable to move OTEL ActiveSupport Notifications subscriber to the front of the notifications list which may cause incomplete traces.' \
+                        'Please report an issue here: ' \
+                        'https://github.com/open-telemetry/opentelemetry-ruby-contrib/issues/new?labels=bug&template=bug_report.md&title=ActiveSupport%20Notifications%20subscribers%20list%20is%20nil'
+            )
           else
-            s = ::ActiveSupport::Notifications.notifier.instance_variable_get(:@subscribers).pop
-            ::ActiveSupport::Notifications.notifier.instance_variable_get(:@subscribers).unshift(s)
+            subscribers.unshift(
+              subscribers.delete(subscriber_object)
+            )
           end
         end
         subscriber_object
       end
 
+      def self.find_subscribers(pattern)
+        active_support_major_version = ::ActiveSupport.version.canonical_segments.first
+
+        # TODO: Drop support for Rails 5 since it is EOL since 2022-06-01
```

Comment:
Do we have an official policy for Rails support? It would be nice to give a grace period beyond the EOL, but it is probably a larger discussion. I would imagine it being something reasonable, like EOL + 1 yr. Whatever we decide, we should have a clear policy.

Reply:
Nothing official. These are all pre 1.0 so they are subject to change at any time. 

Please see this related PR to drop support for Rails 5

https://github.com/open-telemetry/opentelemetry-ruby-contrib/pull/259

Reply:
I 👀 @fbogsany approves. I don't have any strong opinions other than, whatever works for the community works for us. If we drop support and nobody complains, then we're probably fine.